docs/contracts/functions.rst

```diff
@@ -222,6 +222,9 @@ Pure Functions
 --------------
 
 Functions can be declared ``pure`` in which case they promise not to read from or modify the state.
+In particular, it should be possible to evaluate a ``pure`` function at compile-time given
+only its inputs and ``msg.data``, but without any knowledge of the current blockchain state.
+This means that reading from ``immutable`` variables can be a non-pure operation.
```

Comment:
Hmm, should we disallow `immutable` in `pure`? :)

Reply:
Ok, it seems my review was stuck from a few hours ago ^^

But I see that now @ekpyron opened an issue for this.

Reply:
immutables are already view, I think that's what prompted the change here

Reply:
Well than the text here is confusing as it suggests they are still allowed in pure.

Reply:
It does say it's a "non-pure operation", not sure how that's unclear :-)!

Reply:
It says "**can be** a non-pure". So in some cases it "can be pure"?

Reply:
You mean it should have said ``is`` not ``can be`` :-)? Maybe, yeah...

Reply:
Actually, whenever reading an immutable **can be*** pure, it's actually a constant... except that we don't properly evaluate those :-D.